### PR TITLE
Correctly flag JSX elements as being used for the purposes of module imports

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes bug #3674

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -7117,6 +7117,15 @@ namespace ts {
 
             let targetAttributesType = getJsxElementAttributesType(node);
 
+            if(getNodeLinks(node).jsxFlags & JsxFlags.ClassElement) {
+                if(node.tagName.kind === SyntaxKind.Identifier) {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Spaces after the` if`

File: tests/baselines/reference/tsxExternalModuleEmit1.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+//// [tests/cases/conformance/jsx/tsxExternalModuleEmit1.tsx] ////
+
+//// [react.d.ts]
+
+declare module 'react' {
+	class Component<T, U> { }
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/tsxExternalModuleEmit1.js]: Not sure if it matters, but there's a tab here and other files have spaces.

File: tests/baselines/reference/tsxExternalModuleEmit1.errors.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        render() {
+            return <Button />;
+        }
+    
+    }
+    
+==== tests/cases/conformance/jsx/button.tsx (1 errors) ====
+    import * as React from 'react';
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS1148: Cannot compile modules unless the '--module' flag is provided.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/tsxExternalModuleEmit1.errors.txt]: Should the `.tsx` file include a `// @module: commonjs`?

File: src/compiler/checker.ts [hunk 4]
```diff
@@ -7117,6 +7117,15 @@ namespace ts {
 
             let targetAttributesType = getJsxElementAttributesType(node);
 
+            if (getNodeLinks(node).jsxFlags & JsxFlags.ClassElement) {
+                if (node.tagName.kind === SyntaxKind.Identifier) {
+                   checkIdentifier(<Identifier>node.tagName);
+                }
+                else {
+                   checkQualifiedName(<QualifiedName>node.tagName);
```
Comment (JsonFreeman) [inline src/compiler/checker.ts]: Move this one character to the right.

**Comment by prabirshrestha** [comment]
Verified. this fixes the bug.

**Comment by DanielRosenwasser** [comment]
:+1:

**Pull request merged**